earn/src/data/MarketInfo.ts

```diff
@@ -0,0 +1,63 @@
+import Big from 'big.js';
+import { secondsInYear } from 'date-fns';
+import { ethers } from 'ethers';
+import { Address } from 'wagmi';
+
+import { toBig, toImpreciseNumber } from '../util/Numbers';
+
+const RESERVE_FACTOR = 0.825; // (1 - 1/8)
+
+export type MarketInfo = {
+  lender0: Address;
+  lender1: Address;
+  borrowerAPR0: number;
+  borrowerAPR1: number;
+  lender0Utilization: number;
+  lender1Utilization: number;
+  lender0TotalAssets: Big;
+  lender1TotalAssets: Big;
+  lender0TotalBorrows: Big;
+  lender1TotalBorrows: Big;
+  lender0AvailableAssets: Big;
+  lender1AvailableAssets: Big;
+};
+
+export async function fetchMarketInfoFor(
+  lenderLensContract: ethers.Contract,
+  lender0: Address,
+  lender1: Address
+): Promise<MarketInfo> {
+  const [lender0Basics, lender1Basics] = await Promise.all([
+    lenderLensContract.readBasics(lender0),
+    lenderLensContract.readBasics(lender1),
+  ]);
+
+  const interestRate0 = toBig(lender0Basics.interestRate);
+  const borrowAPR0 = interestRate0.eq('0')
+    ? 0
+    : interestRate0.sub(1e12).div(1e12).toNumber() * secondsInYear * RESERVE_FACTOR;
+  const interestRate1 = toBig(lender1Basics.interestRate);
+  const borrowAPR1 = interestRate1.eq('0')
+    ? 0
+    : interestRate1.sub(1e12).div(1e12).toNumber() * secondsInYear * RESERVE_FACTOR;
```

Comment:
```suggestion
  const interestRate0 = toBig(lender0Basics.interestRate);
  const borrowAPR0 = interestRate0.eq('0')
    ? 0
    : interestRate0.sub(1e12).div(1e12).toNumber() * secondsInYear;
  const interestRate1 = toBig(lender1Basics.interestRate);
  const borrowAPR1 = interestRate1.eq('0')
    ? 0
    : interestRate1.sub(1e12).div(1e12).toNumber() * secondsInYear;
```

We don't multiple by `RESERVE_FACTOR` here because this is a borrow rate, not a supply rate.